torch_xla/csrc/translator.cpp

```diff
@@ -152,30 +152,39 @@ XlaTranslator::XlaTranslator(
     const xla::PrecisionConfig::Precision conv_precision)
     : graph_(graph), conv_precision_(conv_precision) {}
 
-xla::XlaComputation XlaTranslator::BuildComputation(
+XlaTranslationResult XlaTranslator::BuildComputation(
     const std::string& name,
     const std::vector<ParameterShape>& parameter_shapes,
+    const std::unordered_map<size_t, XlaComputationInOut::ShapeSizes>&
+        param_size_op_values,
     const BuildOptions& options) const {
   xla::XlaBuilder b(name);
-  auto returned_tuple = BuildComputationProgram(parameter_shapes, &b);
+  auto returned_tuple =
+      BuildComputationProgram(parameter_shapes, param_size_op_values, &b);
   if (options.output_transform) {
     for (size_t i = 0; i < returned_tuple.outputs.size(); ++i) {
       returned_tuple.outputs[i] =
           options.output_transform(returned_tuple.outputs[i], i);
     }
   }
   XlaHelpers::CreateReturnValue(&b, returned_tuple.outputs);
-  return b.Build().ValueOrDie();
+  return {b.Build().ValueOrDie(), returned_tuple.ret_size_op_values};
 }
 
 XlaComputationInOut XlaTranslator::BuildComputationProgram(
     const std::vector<ParameterShape>& parameter_shapes,
+    const std::unordered_map<size_t, XlaComputationInOut::ShapeSizes>&
+        param_size_op_values,
     xla::XlaBuilder* b) const {
   ComputationContext cctx;
   const auto graph_inputs = graph_->inputs();
   XLA_CHECK_EQ(graph_inputs.size(), parameter_shapes.size())
       << "Graph:\n"
       << graph_->toString();
+  // Track the aten::size results while building the XLA computation.
```

Comment:
We have a context now, so let's try not to add more inlined data handling into this growing function ;)